src/merkle/tiered_smt/content.rs

```diff
@@ -0,0 +1,58 @@
+use super::RpoDigest;
+
+#[derive(Debug, Clone, Copy, Eq, PartialEq)]
+pub enum ContentType {
+    Empty,
+    Internal,
+    Leaf,
+}
+
+impl ContentType {
+    pub const fn is_empty(&self) -> bool {
+        matches!(self, Self::Empty)
+    }
+
+    pub const fn is_internal(&self) -> bool {
+        matches!(self, Self::Internal)
+    }
+
+    pub const fn is_leaf(&self) -> bool {
+        matches!(self, Self::Leaf)
+    }
+}
+
+impl Default for ContentType {
+    fn default() -> Self {
+        Self::Empty
+    }
+}
+
+#[derive(Debug, Clone, Default, Eq, PartialEq)]
+pub struct Content {
+    r#type: ContentType,
+    digest: RpoDigest,
+}
```

Comment:
Is the `Content` the right name here? It feels like this described a tree node - so, maybe should be `TreeNode`. Also, I have the same questions as @hackaugusto: could this be an enum?

A few comments on node types:
1. Since empty nodes at the same depth have the same value, should `Empty` node just track the depth? And then we could look up the actual value of an empty node at a given depth in some static array?
2. Technically, empty node is also an internal node - but I agree that it may make sense to track empty nodes separately from non-empty internal nodes.
3. We actually should have another node type: a leaf node at depth 64. This leaf node would contain a sorted list of all leaves which share a common 64-bit prefix for their keys.

Summing up, I think it may be possible to describe a node as:
```Rust
pub enum TreeNode {
    Empty(u8),
    Internal(Word),
    Leaf(Entry),
    LeafList(u64, Vec<Entry>),
}
```
In the above, `Entry` could be a tuple with containing the key and the value which define a leaf.

Reply:
The reason to not use an enum is to detach the need to parse the whole node from the need to peek only the type - something that we do extensively while inserting new keys.

With the current implementation, we just ask the storage for the type, and it will be able to parse a single byte to provide us the required information with the minimal possible overhead. On the other hand, if we use an enum, the storage will have to parse the whole node for every of these iterations, and in most of these cases we will end up not using the parsed information as we will just check if its internal node and traverse until empty/leaf.

wdyt?

Reply:
Make sense! I don't have a good sense of what would work best here (the `TreeNode` enum I proposed above also has some issues). We do need to handle somehow the fact that different types will store different data (i.e., `u8` vs. `Word` vs. `Entry` etc.).